src/main/java/uk/gov/beis/els/categories/lamps/controller/LampsController.java

```diff
@@ -64,22 +69,23 @@ public ModelAndView renderLamps(@ModelAttribute("form") LampsForm form) {
   @PostMapping("/lamps")
   @ResponseBody
   public Object handleLampsSubmit(@Validated @ModelAttribute("form") LampsForm form, BindingResult bindingResult) {
-    if (bindingResult.hasErrors()) {
-      return getLamps(bindingResult.getFieldErrors());
-    }
-    else {
-      return documentRendererService.processPdfResponse(lampsService.generateHtml(form, LampsService.LEGISLATION_CATEGORY_CURRENT));
-    }
+    return doIfValid(form, bindingResult, (category -> documentRendererService.processPdfResponse(lampsService.generateHtml(form, category))));
   }
 
   @PostMapping(value = "/lamps", params = "mode=INTERNET")
   @ResponseBody
   public Object handleInternetLabelLampsSubmit(@Validated(InternetLabellingGroup.class) @ModelAttribute("form") LampsForm form, BindingResult bindingResult) {
+    return doIfValid(form, bindingResult, (category -> documentRendererService.processImageResponse(internetLabelService.generateInternetLabel(form, form.getEfficiencyRating(), category, ProductMetadata.LAMPS_FULL))));
```

Comment:
Think this needs a `ControllerUtils.validateInternetLabelColour` call first, currently the label colour question isn't being validated on the internet label.

Reply:
Oh yes good catch